src/core/Form/MultiSelect/MultiSelect/AriaAnnounceText.tsx

```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+import { AriaLiveMode } from '../../types';
+import { VisuallyHidden } from '../../../VisuallyHidden/VisuallyHidden';
+
+const debounce = (func: Function, waitFor: number) => {
```

Comment:
This whole thing seems to be overly complicated by a quickly looking. Is there some reasoning for having this kind of somewhat complex structure instead of:
- state with timeoutId
- getDerivedStateFromProps to reset timeout and re-apply it when necessary
- Function for the timeout to set announce text state
- Using announce text from state for render